src/components/input/input.js

```diff
@@ -76,11 +76,18 @@ function mdInputContainerDirective($mdTheming, $parse) {
 
   var INPUT_TAGS = ['INPUT', 'TEXTAREA', 'SELECT', 'MD-SELECT'];
 
+  var HIDDEN_INPUT_SELECTORS = [
+    '[hidden=""]', '[hidden="true"]', '[type="hidden"]',
+    '[aria-hidden=""]', '[aria-hidden="true"]', '.md-visually-hidden'
+  ];
+
   var LEFT_SELECTORS = INPUT_TAGS.reduce(function(selectors, isel) {
+    isel = isel + ':not(' + HIDDEN_INPUT_SELECTORS.join('):not(') + ')';
```

Comment:
Since you don't need these as an array, I would go ahead and do the `.join('):not(')` when you define the variable so you don't have to do it twice. I think it will also be a tad bit clearer...too many parentheses here :smile: